nextjs-13/src/components/NewsletterItem/index.test.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+import { render, waitFor, screen, fireEvent } from '@testing-library/react';
+
+import requestManager from 'lib/request/manager';
+
+import NewsletterItem from './index';
+
+jest.mock('lib/request/manager');
+
+jest.mock('sweetalert2-react-content', () => () => ({
+  fire: jest.fn(() => Promise.resolve({ isConfirmed: true })),
+}));
+
+describe('ListNewsletter', () => {
+  it('renders the component', async () => {
+    const item = { id: 1, name: 'Newsletter 1' };
+    render(<NewsletterItem item={item} onAfterCloseCallback={null} />);
+
+    expect(screen.getByTestId('newsletter-item')).toBeInTheDocument();
+    expect(screen.getByText(item.name)).toBeInTheDocument();
+    expect(screen.getByTestId('btn-delete')).toBeInTheDocument();
+  });
+
+  it('deletes record and refresh data on delete button click', async () => {
```

Comment:
Not sure, is it better to split this test into 2 case, one for `delete record` and `refresh data` 🤔

```
describe('on delete newsletters')
      it ('delete a newsletter')
      it ('refresh data') # or `call the callback`
```

Reply:
I was not sure about it too, then I tried to split it into 2 cases to see, and I think it indeed looks better 🙈 

Resolved [here](https://github.com/nimblehq/js-framework-benchmark/pull/116/commits/d3c440d36eaac7b7ec84c3101f5eef14ed62fcf4)